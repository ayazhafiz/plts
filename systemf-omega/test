#!/bin/bash

INPUT="$(pwd)/test.input"
GOLDEN="$(pwd)/test.golden"
REAL="$(pwd)/test.real"

if [ "$1" = "-u" ]; then
  ./run "$INPUT" > "$REAL"
  cp "$REAL" "$GOLDEN"
elif [ "$1" = "-d" ]; then
  ./run "$INPUT" > "$REAL"
  diff --color "$GOLDEN" "$REAL"
else
 ./run "$INPUT"
fi
