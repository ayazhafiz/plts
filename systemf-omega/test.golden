Nat :: *
one: Nat
two: Nat
(λX::*. (λx: X. x)): (∀X::*. (X -> X))
(λx: (∀X::*. (X -> X)). x): ((∀X::*. (X -> X)) -> (∀X::*. (X -> X)))
((λy: Nat. y) one): Nat
Pair :: (* => (* => *))
onetwo: Pair Nat Nat
((λpair: Pair Nat Nat. pair) onetwo): Pair Nat Nat
(λF::*. (λS::*. (λfst: F. (λsnd: S. (λRes::*. (λextract: (F -> (S -> Res)). ((extract fst) snd))))))): (∀F::*. (∀S::*. (F -> (S -> (∀Res::*. ((F -> (S -> Res)) -> Res))))))
(λF::*. (λS::*. (λextract: (∀Res::*. ((F -> (S -> Res)) -> Res)). (extract [F] (λf: F. (λs: S. f)))))): (∀F::*. (∀S::*. ((∀Res::*. ((F -> (S -> Res)) -> Res)) -> F)))
((λextract: (∀Res::*. ((Nat -> (Nat -> Res)) -> Res)). (extract [Nat] (λf: Nat. (λs: Nat. f)))) (((λfst: Nat. (λsnd: Nat. (λRes::*. (λextract: (Nat -> (Nat -> Res)). ((extract fst) snd))))) one) two)): Nat
